tools/src/wyvern/tools/typedAST/core/declarations/DefDeclaration.java

```diff
@@ -235,29 +254,82 @@ public Environment extendName(Environment env, Environment against) {
 	@Override
 	public DeclType genILType(GenContext ctx) {
 		List<FormalArg> args = new LinkedList<FormalArg>();
-		for (NameBinding b : argNames) {
+
+        ctx = this.serializeArguments(args, ctx);
+
+		DefDeclType ret = new DefDeclType(getName(), getResultILType(ctx), args);
+		return ret;
+	}
+
+    private GenContext serializeArguments(List<FormalArg> args, GenContext ctx) {
+        if(isGeneric()) {
+            ValueType type = this.genericStructuralType();
+            args.add(new FormalArg(GENERIC_PREFACE, type));
+            ctx = ctx.extend(GENERIC_PREFACE, new Variable(GENERIC_PREFACE), type);
```

Comment:
This is not working for me at all. I'm spending a lot of time on it.
I'm probably going to give it another hour and move on to inference if I still can't resolve this.

What's happening is that the context, in later method calls, can't find the variable `__generic__` (i.e. `GENERIC_PREFACE`). I'm not sure why.